src/grinder/views.py

```diff
@@ -1,7 +1,183 @@
+    }
+    post_request = requests.post(SPOTIFY_TOKEN_URL, data=code_payload)
+
+    # Auth Step 5: Tokens are Returned to Application
+    response_data = json.loads(post_request.text)
+    access_token = response_data["access_token"]
+    refresh_token = response_data["refresh_token"]
+    token_type = response_data["token_type"]
+    expires_in = response_data["expires_in"]
+
+    # Updating the user.access_token
+    user = User.objects.get(username=request.user.username)
+
+    # Update User tokens and expire date
+    user.access_token = access_token
+    user.refresh_token = refresh_token
+    user.expires_in = expires_in
+
+    # Auth Step 6: Use the access token to access Spotify API
+    authorization_header = {"Authorization": "Bearer {}".format(access_token)}
+
+    # Get profile data
+    Current_users_profile_APIendpoint = "{}/me".format(SPOTIFY_API_URL)
+    User_profile_APIendpoint = "{}/users/{}".format(SPOTIFY_API_URL, user.username)
+    profile_response = requests.get(User_profile_APIendpoint, headers=authorization_header)
+    profile_data = json.loads(profile_response.text)
+
+    # top 100 tracks
+    top50_Apiendpoint ="{}/top/tracks".format(Current_users_profile_APIendpoint)
+    top50_response = requests.get(top50_Apiendpoint, headers=authorization_header)
+    top50 = json.loads(top50_response.text)
+    ids = ''
+    for item in top50["items"]:
+        x = item["id"]
+        ids += x + ','
+    top50_audiofeatures_APIendpoint = "{}/audio-features/?ids={}".format(SPOTIFY_API_URL, ids)
+    top50_audiofeatures_response = requests.get(top50_audiofeatures_APIendpoint, headers=authorization_header)
+    top50_audiofeatures = json.loads(top50_audiofeatures_response.text)
+    # for item in top50["audio_features"]:
+    #     x = item["id"]
+    #     valence
+    #     tempo
+    #     speechiness
+    #     popularity
+    #     mode
+    #     loudness
+    #     liveness
+    #     key
+    #     instrumentalness
+    #     energy
+    #     danceability
+    #     acousticness
+    # recommendations_APIendpoint = "{}/recommendations?{}".format(SPOTIFY_API_URL, seed)
+    # recommendations = requests.get(recommendations_APIendpoint , headers=authorization_header)
+
+
+    user_urls = profile_data["external_urls"]
+    user_sp_url = user_urls["spotify"]
+    name = profile_data["display_name"]
+    print(name, user_sp_url)
+
+    # Get user playlist data
+    playlist_api_endpoint = "{}/playlists".format(profile_data["href"])
+    playlists_response = requests.get(playlist_api_endpoint, headers=authorization_header)
+    playlist_data = json.loads(playlists_response.text)
+
+    # create user output
+    output = '''<blockquote class="blockquote">'''
+    for item in playlist_data["items"]:
+        playlist_names = item["name"]
+        herfs = item["external_urls"]
+        playlist_hrefs = herfs["spotify"]
+        output +=  '''<kbd> <a href="'''+ playlist_hrefs +'''">'''+playlist_names+'''</a> </kbd> <br>'''
```

Comment:
this loop should be in the template itself. In the view we should not parse html output. We just validate and prepare response data. Template is the one who is parsing and generating html content from this data.